employees/tests/test_views.py

```diff
@@ -939,3 +838,157 @@ def _get_total_hours_per_project_and_percentage_from_month(self):
+        )
+        self.data = {
+            "date": datetime.datetime.now().date(),
+            "description": "Some description",
+            "project": self.report.project.pk,
+            "author": self.user.pk,
+            "work_hours": "8:00",
+            "task_activities": self.task_activity.pk,
+        }
+        self.invalid_data = {
+            "date": datetime.datetime.now().date(),
+            "description": "",
+            "project": self.report.project.pk,
+            "author": self.user.pk,
+            "work_hours": "8:00",
+            "task_activities": self.task_activity.pk,
+        }
+
+    def test_create_report_view_should_add_new_report_on_post_when_form_is_valid(self):
+        response = self.client.post(self.url, self.data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Report.objects.all().count(), 2)
+        self.assertEqual(response.url, self.success_url)
+
+    def test_create_report_view_should_not_redirect_and_create_report_on_submit_when_form_is_invalid(self):
+        response = self.client.post(self.url, self.invalid_data)
+        self.assertEqual(Report.objects.all().count(), 1)
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(response, TemplateResponse)
+
+    @parameterized.expand([(2019, 5, 1), (2019, 12, 31), (2020, 2, 29)])
+    def test_create_report_form_default_date_should_be_current_date(self, year, month, day):
+        with freeze_time("{:d}-{:02d}-{:02d}".format(year, month, day)):
+            response = self.client.get(reverse("create-report", kwargs={"year": year, "month": month}))
+        self.assertEqual(
+            response.context_data["form"].initial["date"], timezone.datetime(year=year, month=month, day=day)
+        )
+
+    def test_create_report_form_task_activities_choice_should_be_list_of_activities_specified_for_project(self):
+        not_default_task_activity = TaskActivityTypeFactory()
+        self.assertNotIn(not_default_task_activity, self.report.project.project_activities.all())
+
+        response = self.client.get(self.url, data={"project": self.report.project.pk})
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.context["form"].initial["project"], self.report.project)
+        self.assertIn(self.task_activity, response.context["form"].fields["task_activities"].queryset)
+        self.assertNotIn(not_default_task_activity, response.context["form"].fields["task_activities"].queryset)
+
+    def test_create_report_form_view_should_not_add_report_when_task_activity_is_not_related_not_project(self):
+        not_related_to_project_task_activity = TaskActivityTypeFactory()
+        author_reports_before_post = self.user.report_set.all()
+
+        self.data["task_activities"] = not_related_to_project_task_activity.pk
+
+        response = self.client.post(self.url, self.data)
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
Again - 200 is not a error code. Please fix that

Reply:
It’s also already discussed.